tests/test_postgen_dependencies.py

```diff
@@ -179,6 +179,33 @@ def test_postgenerationmethodcall_fixture(foo: Foo):
     assert foo.number == 456
 
 
+class TestPostgenerationCalledOnce:
+    @register(_name="collector")
+    class CollectorFactory(factory.Factory):
+        class Meta:
+            model = dict
+
+        foo = factory.PostGeneration(lambda *args, **kwargs: 42)
+
+        @classmethod
+        def _after_postgeneration(
+            cls, obj: dict[str, Any], create: bool, results: dict[str, Any] | None = None
+        ) -> None:
+            obj.setdefault("_after_postgeneration_calls", []).append((obj, create, results))
+
+    def test_postgeneration_called_once(self, request):
+        """Test that ``_after_postgeneration`` is called only once."""
+        foo = request.getfixturevalue("collector")
```

Comment:
why not add `collector` as a test's function argument to let `pytest` do the job of getting the fixture value?

Reply:
it's just in case there is an error in the setup, it's nicer to have the tests error out at runtime, rather than at collection time (especially if doing TDD), as it would prevent you from running other tests